### DISCUSSION TITLE
Multiple Rollup Control Records

### DISCUSSION THREAD

**Discussion by louisskelton** [jamessimone/apex-rollup#522] [Q&A]
Hello!

I was wondering if there were any particular considerations around using multiple Rollup Control records. I'd like some processes to be processed synchronously and others in an async queueable context. Initially it made sense for me to create 2x Rollup Control records for this purpose. In my testing this sometimes breaks things... for example if a child object has 2x rollups - one synchronous and one queueable (different parent objects) the queueable rollup won't fire.

I'm guessing from this that we should ensure that our Rollups aren't distributed across Rollup Controls such that child objects are used by more than one Rollup Control - is that right? And if so, what other rules / guidance should we follow when structuring the Rollup Control records and their associated Rollups?

Thanks so much :)

**Comment by jamessimone** [comment]
The two Rollup controls (one sync, one async) should work - let me pull on that thread and see if there's any issue. I haven't tested that with each rollup going to a different parent, but at the moment I don't remember the behavior that determines how to roll children up having logic that operates on a per-parent basis. Let me check on that and get back to you!

**Reply by louisskelton** [reply]
Thanks @jamessimone! I was seeing the same thing where there were two rollups pointing to the same parent too (where one is queueable and the other synchronous). I'm on v1.6.

**Reply by jamessimone** [reply]
@louisskelton I've pushed up a fix for this with an issue for you to track: #525. This will go out in the next version!
